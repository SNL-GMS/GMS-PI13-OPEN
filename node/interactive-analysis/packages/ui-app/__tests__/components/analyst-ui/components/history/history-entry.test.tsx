import Enzyme from 'enzyme';
import React from 'react';
import {
  GenericHistoryEntry,
  makeKeyHandler,
  makeMouseHandler
} from '../../../../../src/ts/components/analyst-ui/components/history/history-stack/history-entry';
import {
  GenericHistoryEntryProps,
  HistoryEntryAction
} from '../../../../../src/ts/components/analyst-ui/components/history/history-stack/types';

describe('history entry', () => {
  const history: GenericHistoryEntryProps = {
    message: 'test',
    entryType: HistoryEntryAction.undo,
    isChild: false,
    isAssociated: true,
    isOrphaned: false,
    isInConflict: false,
    isEventReset: false,
    isAffected: false,
    handleMouseOut: undefined,
    handleAction: undefined,
    handleKeyDown: undefined,
    handleKeyUp: undefined,
    handleMouseEnter: undefined
  };
  test('renders correctly', () => {
    const wrapper = Enzyme.shallow(<GenericHistoryEntry {...history} />);
    expect(wrapper).toMatchSnapshot();
  });
  test('renders correctly', () => {
    const wrapper = Enzyme.shallow(
      <GenericHistoryEntry
        {...history}
        isAssociated={false}
        isInConflict={true}
        entryType={HistoryEntryAction.redo}
        isAffected={true}
        isEventReset={true}
        isOrphaned={true}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  test('mouse in and out handlers work', () => {
    const handler = jest.fn();
    const handledHandler = makeMouseHandler(handler, true);
    const fauxEvent = {
      currentTarget: {
        focus: jest.fn()
      }
    };
    handledHandler(fauxEvent);
    expect(fauxEvent.currentTarget.focus).toHaveBeenCalled();
  });
  test('key handlers work', () => {
    const handler = jest.fn();
    const handledHandler = makeKeyHandler(handler);
    const fauxEvent = {
      currentTarget: {
        focus: jest.fn()
      }
    };
    handledHandler(fauxEvent);
    expect(handler).toHaveBeenCalled();
  });
});
