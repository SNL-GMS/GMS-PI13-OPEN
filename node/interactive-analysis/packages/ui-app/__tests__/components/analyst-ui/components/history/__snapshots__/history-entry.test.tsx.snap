// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`history entry renders correctly 1`] = `
<div
  className="
      list__column history-entry
      history-entry--undo
      is-associated
      
      
      
      
      
    "
  onClick={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  tabIndex={0}
>
  <span
    className="history-entry__description"
  >
    test
  </span>
  <Blueprint3.Tooltip
    content="Undo this action"
    hoverCloseDelay={0}
    hoverOpenDelay={750}
    position="bottom"
    transitionDuration={100}
  >
    <Blueprint3.Icon
      className="history-entry__icon"
      icon="undo"
    />
  </Blueprint3.Tooltip>
</div>
`;

exports[`history entry renders correctly 2`] = `
<div
  className="
      list__column history-entry
      history-entry--redo
      
      is-affected
      
      is-orphaned
      
      is-event-reset
    "
  onClick={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  tabIndex={0}
>
  <span
    className="history-entry__description"
  >
    test
  </span>
  <Blueprint3.Tooltip
    content="This action created a conflict"
    hoverCloseDelay={0}
    hoverOpenDelay={750}
    position="bottom"
    transitionDuration={100}
  >
    <Blueprint3.Icon
      className="history-entry__icon"
      icon="issue"
      intent="danger"
    />
  </Blueprint3.Tooltip>
  <Blueprint3.Tooltip
    content="Restore event history"
    hoverCloseDelay={0}
    hoverOpenDelay={750}
    position="bottom"
    transitionDuration={100}
  >
    <Blueprint3.Icon
      className="history-entry__icon"
      icon="redo"
    />
  </Blueprint3.Tooltip>
</div>
`;
